tensorflow/core/kernels/tensor_to_hash_bucket_op_gpu.cu.cc

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#if GOOGLE_CUDA
+
+#define EIGEN_USE_GPU
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/kernels/tensor_to_hash_bucket_op.h"
+#include "third_party/farmhash_gpu/src/farmhash_gpu.h"
+#include "tensorflow/core/util/gpu_kernel_helper.h"
+
+namespace tensorflow {
+
+namespace {
+
+constexpr int BUFFER_SIZE = 64;
+
+template<typename T>
+__device__ void FillDigits(T val, int num_digits, int* i, char *buf) {
+  eigen_assert(num_digits <= BUFFER_SIZE - (*i));
+
+  int factor = (val < 0 ? -1: 1);
+
+  int num_digits_a = num_digits;
+  while (num_digits > 0) {
```

Comment:
Why can't we check `val != 0` here (i.e. use the same structure as the loop in `IntegerToString`)?

Reply:
Done.